src/server/server.ts

```diff
@@ -152,33 +166,51 @@ export class CTProtoServer<AuthRequestPayload, AuthData, ApiRequest extends NewM
    * @param data - message data
    */
   private async onmessage(socket: ws, data: ws.Data): Promise<void> {
-    try {
-      MessageValidator.validateMessage(data as string);
-    } catch (error) {
-      const errorMessage = (error as Error).message;
-
-      this.log(`Wrong message accepted: ${errorMessage} `, data);
-
-      if (error instanceof CriticalError) {
-        socket.close(CloseEventCode.UnsupportedData, errorMessage);
-      } else {
-        socket.send(MessageFactory.createError('Message Format Error: ' + errorMessage));
-      }
 
-      return;
-    }
+    await this.parseIfString(data, socket);
```

Comment:
where is the return value of parsing?

Reply:
and what does it mean `if string`? it's better to use `if text message` or something similar